wallet-js/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+mod utils;
+
+use chain_impl_mockchain::block::Block;
+use chain_ser::mempack::Readable;
+use wasm_bindgen::prelude::*;
+
+// When the `wee_alloc` feature is enabled, use `wee_alloc` as the global
+// allocator.
+#[cfg(feature = "wee_alloc")]
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+#[wasm_bindgen]
+pub struct Wallet {
+    daedalus: wallet::RecoveringDaedalus,
+    icarus: wallet::RecoveringIcarus,
+    account: wallet::Wallet,
+}
+
+#[wasm_bindgen]
+pub struct Settings(wallet::Settings);
+
+#[wasm_bindgen]
+impl Wallet {
+    /// retrieve a wallet from the given mnemonics and password
+    ///
+    /// this function will work for all yoroi, daedalus and other wallets
+    /// as it will try every kind of wallet anyway
+    ///
+    /// You can also use this function to recover a wallet even after you have
+    /// transferred all the funds to the new format (see the _convert_ function)
+    ///
+    ///
+    /// the mnemonics should be in english
+    pub fn recover(mnemonics: &str, password: &str) -> Result<Wallet, JsValue> {
```

Comment:
password can be an array of bytes for what we care here. can we have `&[u8]` instead?

Reply:
of course, I changed it